readthedocs/notifications/storages.py

```diff
@@ -45,12 +45,12 @@ def _get(self, *args, **kwargs):
         return safe_messages, all_ret
 
     def add(self, level, message, extra_tags='', *args, **kwargs):
-        if self.request.user.is_authenticated():
-            persist_messages = (PersistentMessage.objects
-                                .filter(message=message,
-                                        user=self.request.user,
-                                        read=False))
-            if persist_messages.exists():
-                return
+        user = kwargs.get('user') or self.request.user
```

Comment:
Thinking about this more, there needs to be additional logic. Adding persistent messages, the authentication check does nothing, as we're specifying a user that isn't authed. I think without this, however, an exception would be thrown in the case that we're adding a message to an anonymous user, if we even ever do that, as the authentication check is what prohibits that.